tools/eval/datasets/otb100.py

```diff
@@ -0,0 +1,529 @@
+            yield self.videos[key]
+
+def get_axis_aligned_bbox(region):
+    """Converts a region to (cx, cy, w, h) representing an axis-aligned box."""
+    nv = region.size
+    if nv == 8:
+        cx = np.mean(region[0::2])
+        cy = np.mean(region[1::2])
+        x1 = min(region[0::2])
+        x2 = max(region[0::2])
+        y1 = min(region[1::2])
+        y2 = max(region[1::2])
+        A1 = np.linalg.norm(region[0:2] - region[2:4]) * np.linalg.norm(region[2:4] - region[4:6])
+        A2 = (x2 - x1) * (y2 - y1)
+        s = np.sqrt(A1 / A2)
+        w = s * (x2 - x1) + 1
+        h = s * (y2 - y1) + 1
+    else:
+        x, y, w, h = region
+        cx = x + w / 2
+        cy = y + h / 2
+    return cx, cy, w, h
+
+def load_text_numpy(path, delimiter, dtype):
+    for d in delimiter:
+        try:
+            ground_truth_rect = np.loadtxt(path, delimiter=d, dtype=dtype)
+            return ground_truth_rect
+        except:
+            pass
+
+    raise Exception('Could not read file {}'.format(path))
+
+class OTB100:
+    def __init__(self, root):
+        # Go up one if directory is provided
+        root = os.path.abspath(root)
+        if root.endswith("OTB100"):
+            root = os.path.dirname(root)
+
+        self.dataset = OTBDATASET(root)
+
+    @property
+    def name(self):
+        return self.__class__.__name__
+
+    def eval(self, model):
+        for v_idx, video in enumerate(self.dataset):
+            total_time = 0
+            pred_bboxes = []
+            scores = []
+            track_times = []
+
+            for idx, (img, gt_bbox) in enumerate(video):
+                img = cv.cvtColor(img, cv.COLOR_BGR2RGB)
+                tic = cv.getTickCount()
+
+                if idx == 0:
+                    cx, cy, w, h = get_axis_aligned_bbox(np.array(gt_bbox))
+                    gt_bbox_ = (int(cx - w / 2), int(cy - h / 2), int(w), int(h))
+                    model.init(img, gt_bbox_)
+                    pred_bbox = gt_bbox_
+                    scores.append(None)
+                else:
+                    isLocated, bbox, score = model.infer(img)
+                    pred_bbox = bbox
+                    scores.append(score)
+
+                pred_bboxes.append(pred_bbox)
+                toc = (cv.getTickCount() - tic) / cv.getTickFrequency()
+                total_time += toc
+                track_times.append(toc)
+
+            model_path = os.path.join('OTB_results')
+            os.makedirs(model_path, exist_ok=True)
+            result_path = os.path.join(model_path, '{}.txt'.format(video.name))
+            print(result_path)
+            with open(result_path, 'w') as f:
+                for bbox in pred_bboxes:
+                    f.write(','.join(map(str, bbox)) + '\n')
+
+            avg_fps = len(video) / total_time if total_time > 0 else 0
+            print('({:3d}) Video: {:12s} Time: {:5.1f}s Speed: {:3.1f}fps'.format(
+                v_idx + 1, video.name, total_time, avg_fps))
```

Comment:
It prints too many lines. Could you make it a progress bar in the outer loop (line 277) so that It can print something close to the following?

```
Evaluating: 30%|█████                  | 30/100 [00:00<12:34, Woman]
```

Reply:
Updated to print a progress bar in the outer loop :+1: 
![image](https://github.com/opencv/opencv_zoo/assets/43348991/93f4490a-9932-4799-b253-061a15074fdf)